File: src/main/templates/javascript_sdk/src/example/UserApi.js
import { User } from './User.js';
import { PageResult } from '../api/PageResult.js';

/**
 * User API - 直接访问User实体（示例）
 * 示例用法：
 * const userApi = new UserApi(apiClient, "myDatasource", "user");
 * const users = await userApi.listUsers();
 * const user = await userApi.getUser("123");
 */
export class UserApi {
  constructor(apiClient, datasourceName, modelName) {
    this.apiClient = apiClient;
    this.datasourceName = datasourceName;
    this.modelName = modelName;
  }

  /**
   * 获取所有User记录（分页）
   */
  async listUsers(options = {}) {
    const {
      current = null,
      pageSize = null,
      filter = null,
      nestedQuery = null,
      sort = null
    } = options;

    const path = `/api/f/datasources/${this.encode(this.datasourceName)}/models/${this.encode(this.modelName)}/records`;
    const params = {};
    
    if (current !== null) params.current = current;
    if (pageSize !== null) params.pageSize = pageSize;
    if (filter !== null) params.filter = filter;
    if (nestedQuery !== null) params.nestedQuery = nestedQuery;
    if (sort !== null) params.sort = sort;

    const data = await this.apiClient.get(path, params);
    const users = data.list.map(item => User.fromJSON(item));
    return new PageResult(data.total, users);
  }

  /**
   * 获取所有User记录（直接返回列表）
   * 这是您想要的方法：const users = await userApi.listUsers();
   */
  async listUsersAsList(options = {}) {
    const pageResult = await this.listUsers(options);
    return pageResult.getList();
  }

  /**
   * 获取所有User记录（最简单的调用）
   */
  async listUsersSimple() {
    return await this.listUsersAsList();
  }

  /**
   * 根据ID获取User记录
   */
  async getUser(id, options = {}) {
    const { nestedQuery = null } = options;
    const path = `/api/f/datasources/${this.encode(this.datasourceName)}/models/${this.encode(this.modelName)}/records/${this.encode(id)}`;
    const params = {};
    
    if (nestedQuery !== null) params.nestedQuery = nestedQuery;

    const data = await this.apiClient.get(path, params);
    return User.fromJSON(data);
  }

  /**
   * 创建新的User记录
   */
  async createUser(user) {
    const path = `/api/f/datasources/${this.encode(this.datasourceName)}/models/${this.encode(this.modelName)}/records`;
    const data = await this.apiClient.post(path, user.toJSON());
    return User.fromJSON(data);
  }

  /**
   * 更新User记录
   */
  async updateUser(id, user) {
    const path = `/api/f/datasources/${this.encode(this.datasourceName)}/models/${this.encode(this.modelName)}/records/${this.encode(id)}`;
    const data = await this.apiClient.put(path, user.toJSON());
    return User.fromJSON(data);
  }

  /**
   * 部分更新User记录
   */
  async patchUser(id, user) {
    const path = `/api/f/datasources/${this.encode(this.datasourceName)}/models/${this.encode(this.modelName)}/records/${this.encode(id)}`;
    const data = await this.apiClient.patch(path, user.toJSON());
    return User.fromJSON(data);
  }

  /**
   * 删除User记录
   */
  async deleteUser(id) {
    const path = `/api/f/datasources/${this.encode(this.datasourceName)}/models/${this.encode(this.modelName)}/records/${this.encode(id)}`;
    await this.apiClient.delete(path);
  }

  /**
   * URL编码
   */
  encode(str) {
    return str.replace(/\//g, '%2F');
  }
}

File: src/main/templates/python_sdk/src/example/user_api.py
"""
User API - 直接访问User实体（示例）
"""

from typing import List, Optional, Dict, Any
from ..client.api_client import ApiClient
from ..api.page_result import PageResult
from .user import User


class UserApi:
    """User API - 直接访问User实体（示例）"""
    
    def __init__(self, api_client: ApiClient, datasource_name: str, model_name: str):
        """
        初始化User API
        
        Args:
            api_client: API客户端实例
            datasource_name: 数据源名称
            model_name: 模型名称
        """
        self.api_client = api_client
        self.datasource_name = datasource_name
        self.model_name = model_name
    
    def _encode(self, text: str) -> str:
        """URL编码"""
        return text.replace('/', '%2F')
    
    def list_users(
        self,
        current: Optional[int] = None,
        page_size: Optional[int] = None,
        filter_expr: Optional[str] = None,
        nested_query: Optional[bool] = None,
        sort: Optional[str] = None
    ) -> PageResult[User]:
        """
        获取所有User记录（分页）
        
        Args:
            current: 当前页
            page_size: 页大小
            filter_expr: 过滤条件
            nested_query: 是否嵌套查询
            sort: 排序
            
        Returns:
            分页结果
        """
        path = f"/api/f/datasources/{self._encode(self.datasource_name)}/models/{self._encode(self.model_name)}/records"
        
        params = {}
        if current is not None:
            params['current'] = current
        if page_size is not None:
            params['pageSize'] = page_size
        if filter_expr is not None:
            params['filter'] = filter_expr
        if nested_query is not None:
            params['nestedQuery'] = nested_query
        if sort is not None:
            params['sort'] = sort
        
        data = self.api_client.get(path, params)
        users = [User.from_dict(item) for item in data['list']]
        return PageResult(data['total'], users)
    
    def list_users_as_list(
        self,
        filter_expr: Optional[str] = None,
        nested_query: Optional[bool] = None,
        sort: Optional[str] = None
    ) -> List[User]:
        """
        获取所有User记录（直接返回列表）
        这是您想要的方法：users = user_api.list_users_as_list()
        
        Args:
            filter_expr: 过滤条件
            nested_query: 是否嵌套查询
            sort: 排序
            
        Returns:
            User对象列表
        """
        page_result = self.list_users(
            filter_expr=filter_expr,
            nested_query=nested_query,
            sort=sort
        )
        return page_result.get_items()
    
    def list_users_simple(self) -> List[User]:
        """
        获取所有User记录（最简单的调用）
        
        Returns:
            User对象列表
        """
        return self.list_users_as_list()
    
    def get_user(
        self,
        user_id: str,
        nested_query: Optional[bool] = None
    ) -> User:
        """
        根据ID获取User记录
        
        Args:
            user_id: 用户ID
            nested_query: 是否嵌套查询
            
        Returns:
            User对象
        """
        path = f"/api/f/datasources/{self._encode(self.datasource_name)}/models/{self._encode(self.model_name)}/records/{self._encode(user_id)}"
        
        params = {}
        if nested_query is not None:
            params['nestedQuery'] = nested_query
        
        data = self.api_client.get(path, params)
        return User.from_dict(data)
    
    def create_user(self, user: User) -> User:
        """
        创建新的User记录
        
        Args:
            user: User对象
            
        Returns:
            创建的User对象
        """
        path = f"/api/f/datasources/{self._encode(self.datasource_name)}/models/{self._encode(self.model_name)}/records"
        data = self.api_client.post(path, user.to_dict())
        return User.from_dict(data)
    
    def update_user(self, user_id: str, user: User) -> User:
        """
        更新User记录
        
        Args:
            user_id: 用户ID
            user: User对象
            
        Returns:
            更新后的User对象
        """
        path = f"/api/f/datasources/{self._encode(self.datasource_name)}/models/{self._encode(self.model_name)}/records/{self._encode(user_id)}"
        data = self.api_client.put(path, user.to_dict())
        return User.from_dict(data)
    
    def patch_user(self, user_id: str, user: User) -> User:
        """
        部分更新User记录
        
        Args:
            user_id: 用户ID
            user: User对象
            
        Returns:
            更新后的User对象
        """
        path = f"/api/f/datasources/{self._encode(self.datasource_name)}/models/{self._encode(self.model_name)}/records/{self._encode(user_id)}"
        data = self.api_client.patch(path, user.to_dict())
        return User.from_dict(data)
    
    def delete_user(self, user_id: str) -> None:
        """
        删除User记录
        
        Args:
            user_id: 用户ID
        """
        path = f"/api/f/datasources/{self._encode(self.datasource_name)}/models/{self._encode(self.model_name)}/records/{self._encode(user_id)}"
        self.api_client.delete(path)
